### PR TITLE
feat: add retainCodecs and retainCandidates functions

### PR DESCRIPTION
**Pull request by brycetham**
This PR adds the `retainCodecs` and `retainCandidates` functions to `munge.ts`, as part of [SPARK-411372](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-411372). These functions can accept either a parsed `Sdp` object or an `AvMediaDescription` object.

I have also updated `removeCodec` to also accept either an `Sdp` or an `AvMediaDescription`.

### CODE REVIEW

File: src/munge.ts [hunk 1]
```diff
@@ -42,17 +42,71 @@ export function disableRemb(sdp: Sdp) {
...
+ * Filter out unwanted codecs from the given SDP or audio/video media description.
+ *
+ * Note: Done this way because of a feature not implemented in all browsers, currently missing in
+ * Firefox. Once that is added we can use `RTPSender.getCapabilities` and filter those to call
+ * with `RTCRtpTransceiver.setCodecPreferences` instead of doing this manually.
+ *
+ * @param sdpOrAv - The {@link Sdp} or {@link AvMediaDescription} from which to filter codecs.
+ * @param allowedCodecNames - The names of the codecs that should remain in the SDP.
+ */
+export function filterCodecs(
```
Comment (bbaldino) [inline src/munge.ts:70]: Wondering if we can name this something better...I feel like there's ambiguity around if the passed list are the ones being kept or the ones being removed.

Comment (bbaldino) [reply src/munge.ts:70]: `retainCodecs` maybe

**Review by bbaldino — APPROVED** [review]
looks good, one thought on naming but not a huge deal

File: src/munge.ts (hunk 1 above)
Comment (brycetham) [reply src/munge.ts:70]: Sure, I can also rename `filterCandidates` to `retainCandidates`.

**Review by bbaldino — APPROVED** [review]

**Pull request merged**